### ISSUE TITLE
[Feature Requirement] lint props passed to a `React.memo` component

### ISSUE THREAD

**Issue by indooorsman** [pmndrs/eslint-plugin-valtio#37]
as descripted here: https://github.com/pmndrs/valtio/wiki/Some-gotchas#using-reactmemo-with-object-props-may-result-in-unexpected-behavior, a rule to check if a prop of `Reac.memo` component is `snapshot` would be very helpful

**Comment by dai-shi** [comment]
It would be super nice, but I'm not sure if it's possible. Can we know if a component is wrapped by `React.memo`?
cc: @barelyhuman 

---

Hm, the rule can't even detect with `useMemo`?
```js
// invalid case
const Component = () => {
  const snap = useSnapshot(state)
  useMemo(() => ..., [snap]) // invalid
  ...
```
(Maybe, we discussed this once.)

**Comment by barelyhuman** [comment]
We can add `useMemo` pretty easily, thought `React.memo` might conflict with existing checks, so I'll try it out and let you know

**Comment by dai-shi** [comment]
Valtio v2 is released and it doesn't have this gotcha any more.
